### ISSUE TITLE
Overset mesh issue

### ISSUE THREAD

**Issue by tianboxi** [mdolab/adflow#132]
## Description
Hi,

I have encountered an issue when using the overset mesh feature for a simulation of a flapped airfoil. My mesh is shown in the following picture (with 80% transparency) with only computed (iBlank=1) cells turned on. It consists of three blocks: The background mesh (green), component mesh #1 for the main airfoil element (grey) and component mesh #2 for the flap(red).  (yellow part is due to the overlap of green and red)
![Computed_Cells](https://user-images.githubusercontent.com/32691862/111236577-72b2f080-85b0-11eb-92ea-698fa69352bc.png)

## Current behavior 
The issue is that some cells are taken as fringe cells mistakenly thus missing from the computed cell. As you can see in the following zoomed in picture showing that some cells are missing at the top and near the trailing edge of the main airfoil element. 
![gap_zoomed](https://user-images.githubusercontent.com/32691862/111237420-44361500-85b2-11eb-8d61-7be98a5f1602.png)

The fringe cells (iblank=-1) is shown in the next picture in which you can see the cells missing from the computed cells shows up here.
![Fringe_wzoom](https://user-images.githubusercontent.com/32691862/111237535-82cbcf80-85b2-11eb-9de3-01256d8783da.png)

## Expected behavior
All the cells near the wall boundary of the main airfoil element should be taken as computed cells

Could you advice how can I fix this issue by either changing some parameters in the settings or altering the mesh in a certain way? Any help would be greatly appreciated! 

### Code versions
Adflow - V2.4.0

**Comment by anilyil** [comment]
Hi @tianboxi, you can try setting `oversetPriority` in your options dictionary. You should pass a dictionary here. The keys will be the block IDs (the full name of the block), and the values will set the "priority". A lower priority is better, so if you set a lower value for a block, the code will prefer to pick the cells from that block that many times more. The code really tries to pick the smaller cells in overlapped regions, by setting a "low" priority, you are basically setting a multiplier for that block's cell volumes so they appear smaller. The option will look like this:

```
"oversetPriority": { 
    "block_on_wall_1":0.001,
    "block_on_wall_2":0.001},
```

**Comment by tianboxi** [comment]
Hi @anilyil, thanks a lot for the suggestion. I've tried using the `oversetPriority` option before and it didn't help. The cells that are missing are actually already the smallest cells in that overlapped region. Thus applying a multiplier to lower the cell volume didn't work. The problem is that the missing cells were recognized as interpolated cells (iblank=-1) rather than computed cells (iblank=1).

**Comment by anilyil** [comment]
I understand that they were selected as interpolate cells. The algorithm that determines if a cell is a compute/interpolate/blank etc. uses that overset priority stuff. So you can modify the oversetpriority to change the outcome of the algorithm that determines the iblank value. 

If the code set those cells as interpolate cells, then it is picking compute cells around the same area. I still think reducing the oversetpriority further should help. Also, maybe you set the block IDs incorrectly?

**Comment by tianboxi** [comment]
I am still unable to solve the issue. It seems that the algorithm simply missed that part. There's no compute cell from any of the block covering that missing part.  
![Screenshot from 2021-03-16 15-22-42](https://user-images.githubusercontent.com/32691862/111387863-a6068580-866b-11eb-9823-d3602892ad83.png)
Also from this following view. The red cells are clearly larger than the green, yet still being selected as compute cell. (this is without overset priority options applied)
![Screenshot from 2021-03-16 15-23-05](https://user-images.githubusercontent.com/32691862/111388100-fd0c5a80-866b-11eb-9064-c9e641318405.png)

**Comment by anilyil** [comment]
I think there is a default scaling in the algorithm for comparing volumes of an overlapping mesh so a smaller size does not always guarantee that the overlapping mesh will be blanked.

I have not ran into issues like this before. I think you will probably need to modify the mesh slightly. I am not sure what is causing the issue. Especially the boundary nodes getting blanked looks wrong like you said.

**Comment by anilyil** [comment]
Have you tried setting the `nearwalldist` option? I think it is used to determine how the off-wall cells are preserved as compute cells. Default value of 0.01 makes sense for metric units, but if your grid is in say inches, you will need a larger value.

**Comment by tianboxi** [comment]
After trying modifying the meshes and play around with `oversetPriority`, I was able to get all the missing cells at the wall boundary back to compute node. But there are still some cells missing near wall. Then I tried changing the `nearWallDist` value as you suggested and I found that strangely reducing the value helped recover all the rest missing cells. Overall, I think the problem is mostly caused by the closely spaced geometry. The two components (main element and the flap) are too close together and the thus fringes are too close to the wall boundary.  Thank you for the help! @anilyil
